src/components/ProfileContent.tsx

```diff
@@ -0,0 +1,172 @@
+ * @constructor
+ */
+const UserRegistration: FC<{ id: string; status: string }> = ({ id, status }) => {
+    const { t } = useTranslation("Profile");
+    const { t: tStatus } = useTranslation("Profile", { keyPrefix: "status_names" });
+    const bg = useThemeBackground("secondary");
+    return (
+        <View style={[bg, styles.pill]}>
+            <Label type="strong" color="white">
+                {t("registration_nr")} {id} | {tStatus(status)}
+            </Label>
+        </View>
+    );
+};
+
+export type ProfileContentProps = {
+    claims: Claims;
+    user: UserRecord;
+    parentPad?: number;
+};
+
+/**
+ * User profile page.
+ * @param claims The IDP user claims.
+ * @param user The backend user info.
+ * @param parentPad The padding that the parent will apply.
+ * @constructor
+ */
+export const ProfileContent: FC<ProfileContentProps> = ({ claims, user, parentPad = 0 }) => {
+    const { t } = useTranslation("Profile");
+    const avatarBackground = useThemeBackground("primary");
+    const { logout } = useAuthContext();
+
+    const isAttendee = user.Roles.includes("Attendee");
+    const isCheckedIn = user.Roles.includes("AttendeeCheckedIn");
+    const roleComplex = Boolean(user.Roles.find((role) => role !== "Attendee" && role !== "AttendeeCheckedIn"));
+    return (
+        <>
+            {isCheckedIn ? (
+                <Badge unpad={parentPad} badgeColor="primary" textColor="invText">
+                    {t("roles_simple_checked_in")}
+                </Badge>
+            ) : isAttendee ? (
+                <Badge unpad={parentPad} badgeColor="warning" textColor="invText">
+                    {t("roles_simple_attendee")}
+                </Badge>
+            ) : null}
+            <View style={styles.avatarContainer}>
+                <Image
+                    style={[avatarBackground, styles.avatarCircle]}
+                    source={claims.avatar ?? assetSource("ych")}
+                    contentFit="contain"
+                    placeholder="ych"
+                    transition={60}
+                    cachePolicy="memory"
+                    priority="high"
+                />
+            </View>
+
+            <Label type="h1" variant="middle">
+                {claims.name as string}
+            </Label>
+
+            <Label type="caption" variant="middle" mb={20}>
+                {claims.email as string}
+            </Label>
+
+            <View style={styles.registrations}>
+                {user.Registrations.map((r) => (
+                    <UserRegistration key={r.Id} id={r.Id} status={r.Status} />
+                ))}
+            </View>
+
+            <Label mt={20} type="para">
+                {t("login_description", { conName })}
+            </Label>
+
+            <Button style={styles.idpButton} outline icon="web" onPress={() => Linking.openURL(authSettingsUrl).catch(captureException)}>
+                {t("idp_settings")}
+            </Button>
+
+            {!roleComplex ? null : (
```

Comment:
```suggestion
            {!roleComplex && (
```

You can do the same thing without having to specify the else condition